setup

```diff
@@ -55,81 +81,145 @@ else
+	echo "Generating $BTK_BLOCKS initial blocks for regtest..."
+    # echo "Generating $BTK_BLOCKS blocks for regtest to address"
+    # echo $BTK_GENERATE_ADDRESS
+
+    docker-compose exec -T bitcoind-regtest bitcoin-cli generate $BTK_BLOCKS > /dev/null
+    sleep $BTK_SLEEP_TIME
+    
+	# Stop the server(s)
+	docker-compose exec -T bitcoind-regtest bitcoin-cli stop
+	# docker-compose stop
+
+	# Remove temporary env file
+	# rm ".env"
+
+	# Reset the compose_file
+	# export COMPOSE_FILE="$PRE_NETWORK_FILE"
+    # export COMPOSE_FILE="$COMPOSE_FILE:./dc_files/docker-compose.fulcrum.yml"
 
-# Check if network exists and create if it doesn't
-NETWORK_NAME=bitcoin-bridging
-if [ -z $(docker network ls --filter name=^${NETWORK_NAME}$ --format="{{ .Name }}") ] ; then 
-     docker network create ${NETWORK_NAME} ; 
 fi
 
-# Create new SSL certificate/key pair to be used on nginx server
-if [ -d "./cert" ]; then 
-	if [[ -f "./cert/cert.key" && -f "./cert/cert.crt" ]]; then
-	    echo "OpenSSL certificates already exist."
+if [[ $BTK_NGINX == 1 ]]; then
+
+	# Check that openssl is installed and available
+	echo "Looking for openssl..."
+	if [[ -x "$(command -v openssl)" ]]; then
+		echo "Openssl found."
 	else
-	    echo "Removing previous certs directory"
-	    rm -rf ./cert
+		echo "Openssl not found. Please install openssl."
+		exit 1
 	fi
-fi
 
-if [ ! -d "./cert" ]; then
-	mkdir ./cert
-	echo "Creating SSL certificate to use for nginx reverse proxy server."
-	openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
-		-keyout ./cert/cert.key \
-		-out ./cert/cert.crt \
-		-subj "/C=no/ST=no/L=no/O=no/OU=no/CN=no/emailAddress=no"
-fi
+	# Create new SSL certificate/key pair to be used on nginx server
+	if [[ -d "./configs/cert" ]]; then 
+		if [[ -f "./configs/cert/cert.key" && -f "./configs/cert/cert.crt" ]]; then
+			echo "OpenSSL certificates already exist."
+		else
+			echo "Removing previous certs directory"
+			rm -rf ./configs/cert
+		fi
+	fi
 
+	if [[ ! -d "./configs/cert" ]]; then
+		mkdir ./configs/cert
+		echo "Creating SSL certificate to use for nginx reverse proxy server."
+		openssl req -x509 -nodes -days 365 -newkey rsa:2048 \
+			-keyout ./configs/cert/cert.key \
+			-out ./configs/cert/cert.crt \
+			-subj "/C=no/ST=no/L=no/O=no/OU=no/CN=no/emailAddress=no"
+	fi
 
-if [[ $BTK_NODE == 1 ]]; then
-	echo "Starting local node servers..."
-	docker-compose up -d
-	sleep 5
+	export COMPOSE_FILE="$COMPOSE_FILE:./dc_files/docker-compose.nginx.yml"
 
-	# Generate some blocks
-	docker-compose exec -T bitcoind-regtest bitcoin-cli generate 210
+fi
 
-	# Stop the server(s)
-	docker-compose stop 
+if [[ $BTK_REST == 1 ]]; then
+	# export COMPOSE_FILE="$COMPOSE_FILE:./dc_files/docker-compose.opensight.yml"
+	# export COMPOSE_FILE="$COMPOSE_FILE:./dc_files/docker-compose.restapi.yml"
+	docker pull actorforth/opensight:latest
+	docker pull actorforth/rest.bitcoin.com:latest
+fi
 
-	#
-	touch .BTK-container-lock
+if [[ $BTK_JS_API == 1 ]]; then
```

Comment:
I test the ./setup script and there is not setup bch-api.

Reply:
Yeah, I have an idea for a workaround to make bch-api an option again. I'll add that as an additional commit once I figure something out.